### ISSUE TITLE
Users without join private contest authorizatipm are able to create clarification

### ISSUE THREAD

**Issue by SpaceWhite** [omegaup/omegaup#1932]
## Current behaviour
Users are able to create clarification with private contest_alias and private problem that they don’t have access to

## Expected behaviour
API in clarification controller should also check whether if the user has access to private contest or not

## Possible solution
Call ContestController::canAccessContest function in
https://github.com/omegaup/omegaup/blob/15b0810f6512da9061048e43850ce068f8928df0/frontend/server/controllers/ClarificationController.php#L28

**Comment by SpaceWhite** [comment]
I think it would be better to also move canAccessContest and isInvitedToContest function from ContestController to Authorization lib.
@lhchavez What do you think?

**Comment by lhchavez** [comment]
Yes to everything @SpaceWhite said. You keep finding all sorts of obscure bugs!

Assigned to @dgarvit (but keep in mind you already have two other issues assigned. please prioritize finishing those first).

**Comment by SpaceWhite** [comment]
@dgarvit are you still working on this issue?

**Comment by dgarvit** [comment]
No. Please go ahead! :)
